autogpt_platform/backend/linter.py

```diff
@@ -21,23 +21,31 @@ def run(*command: str) -> None:
         )
     except subprocess.CalledProcessError as e:
         print(e.output.decode("utf-8"), file=sys.stderr)
+        raise
 
 
 def lint():
-    try:
-        run("ruff", "check", *TARGET_DIRS, "--exit-zero")
-        run("ruff", "format", "--diff", "--check", LIBS_DIR)
-        run("isort", "--diff", "--check", "--profile", "black", BACKEND_DIR)
-        run("black", "--diff", "--check", BACKEND_DIR)
-        run("pyright", *TARGET_DIRS)
-    except subprocess.CalledProcessError as e:
-        print("Lint failed, try running `poetry run format` to fix the issues: ", e)
-        raise e
+    lint_step_args: list[list[str]] = [
+        ["ruff", "check", *TARGET_DIRS, "--exit-zero"],
+        ["ruff", "format", "--diff", "--check", LIBS_DIR],
+        ["isort", "--diff", "--check", "--profile", "black", BACKEND_DIR],
+        ["black", "--diff", "--check", BACKEND_DIR],
+        ["pyright", *TARGET_DIRS],
+    ]
+    lint_error = None
+    for args in lint_step_args:
+        try:
+            run(*args)
+        except subprocess.CalledProcessError as e:
+            lint_error = e
+
+    if lint_error:
+        print("Lint failed, try running `poetry run format` to fix the issues")
+        sys.exit(1)
 
 
 def format():
     run("ruff", "check", "--fix", *TARGET_DIRS)
     run("ruff", "format", LIBS_DIR)
     run("isort", "--profile", "black", BACKEND_DIR)
     run("black", BACKEND_DIR)
-    run("pyright", *TARGET_DIRS)
```

Comment:
why remove :(
I use this all the time

Reply:
Lint = check, format = fix. This is a check that is under the `format` command. 

How do you use it? Do you have pre-commit hook set up?

Reply:
from a dev perspective, running pyright is a fixer, because it tells me what I need to fix, if you remove it, please add a command called fix to both the yarn and poetry

Reply:
ok I see, I'll put it back